### ISSUE TITLE
Alt Gr moves the cursor acts like the left arrow key

### ISSUE THREAD

**Issue by omar-polo** [godotengine/godot#51809]
### Godot version

4.0.dev.custom_build.1f94211aa

### System information

OpenBSD, Vulkan

### Issue description

When I press alt gr (aka ISO_Level3_Shift) in the text editor the cursor moves left by one character, just like if the left arrow key was pressed.  The alt gr otherwise acts correctly (i.e. alt gr + something else produces the correct letter).

When I press alt gr `xev` produces the following output:

```
KeyPress event, serial 34, synthetic NO, window 0x4000001,
    root 0x6b8, subw 0x0, time 49682853, (89,89), root:(926,235),
    state 0x0, keycode 113 (keysym 0xfe03, ISO_Level3_Shift), same_screen YES,
    XLookupString gives 0 bytes:
    XmbLookupString gives 0 bytes:
    XFilterEvent returns: False
```

I'm using a custom keyboard layout (a modified dvorak fwiw), but even if I change the layout to `it` with setxkbmap the issue remains.  

I can't reproduce the issue on Godot 3.3.2

### Steps to reproduce

Create a new empty project, then create a script and press and release the alt gr key.

**Comment by akien-mga** [comment]
CC @bruvzg

**Comment by jmb462** [comment]
I can't reproduce the issue under Ubuntu Budgie (nor Windows 10).

```
KeyPress event, serial 35, synthetic NO, window 0x5400001,
    root 0x1e8, subw 0x0, time 1266320, (388,159), root:(429,270),
    state 0x10, keycode 108 (keysym 0xfe03, ISO_Level3_Shift), same_screen YES,
    XKeysymToKeycode returns keycode: 92
    XLookupString gives 0 bytes: 
    XmbLookupString gives 0 bytes: 
    XFilterEvent returns: False
```

**Comment by TechnoPorg** [comment]
> There is something I can do to provide to aid the debugging?

Bisecting is always helpful, if you want to do that!

**Comment by akien-mga** [comment]
Can you still reproduce it in the latest version?

**Comment by omar-polo** [comment]
Yes, it still happens.  I can reproduce it with both a custom keyboard layout (a modification of the dvorak-programmer) and with the "normal" Italian QWERTY layout.  The output from xev(1) is the same as in the first post, and Godot is the only program where I've observed this behaviour.

**Comment by bruvzg** [comment]
Can't reproduce, probably something OpenBSD specific.
